tvm/src/codegen/llvm/codegen_llvm.h

```diff
@@ -279,6 +280,17 @@ class CodeGenLLVM : public ExprFunctor<llvm::Value*(const Expr&)>,
   // for Return
   bool has_return_{false};
 
+  // for Assert
+  struct assert_free {
+    VarExpr buffer_var;
+    Expr dev_type;
+    Expr dev_id;
+    size_t depth;
+  };
+  std::vector<assert_free> assert_alloc_mem;
+  bool save_buffer_flag{false};
+  size_t loop_depth;
```

Comment:
It's not just "loop" depth, right? According to your code, you're finding the branches. I guess more specifically, you're checking the liveness of an allocation? Could you give me an example where the depth is important?

Reply:
Yeah loop depth isn't the best name for it since it's also used for if statements. If there are nested loops or if statements, it only needs to free the memory that was allocated in the same level of the assert statement or lower since anything declared in an inner loop/if statement would have already been freed

Reply:
In that case, can we also check the free call and remove it from your free list? Otherwise I'm afraid this `loop_depth` might not cover all cases.